shared-module/_pixelbuf/PixelBuf.c

```diff
@@ -109,26 +111,31 @@ void common_hal__pixelbuf_pixelbuf_set_brightness(mp_obj_t self_in, mp_float_t b
     pixelbuf_pixelbuf_obj_t* self = native_pixelbuf(self_in);
     // Skip out if the brightness is already set. The default of self->brightness is 1.0. So, this
     // also prevents the pre_brightness_buffer allocation when brightness is set to 1.0 again.
-    mp_float_t change = brightness - self->brightness;
-    if (-0.001 < change && change < 0.001) {
+    self->brightness = brightness;
+    uint16_t new_scaled_brightness = (int)roundf(brightness * 256);
```

Comment:
```suggestion
    // Use 256 steps of brightness so that we can do integer math below.
    uint16_t new_scaled_brightness = (int)roundf(brightness * 256);
```